custom/icds_reports/views.py

```diff
@@ -1949,3 +1949,33 @@ def get_type_code(self, data_type):
             "awc": 3
         }
         return type_map[data_type]
+
+
+@location_safe
+@method_decorator([api_auth, toggles.mwcd_indicators.required_decorator()], name='dispatch')
+class MWCDDataView(View):
+
+    def message(self, message_name):
+        error_messages = {
+            "unknown_error": "Unknown Error occured",
+            "no_data": "Data does not exists"
+        }
+
+        return error_messages[message_name]
+
+    def get(self, request, *args, **kwargs):
+
+        try:
+            data = get_mwcd_indicator_api_data()
+            response = {'isSuccess': True,
+                        'message': 'Data Sent Successfully',
+                        'Result': {
+                            'response': data
+                        }}
+            return JsonResponse(response)
+        except MWCDReportView.DoesNotExist:
+            response = dict(isSuccess=False, message=self.message('no_data'))
+            return JsonResponse(response, status=500)
+        except AttributeError:
```

Comment:
Same question about this one. You seem to be catching the same errors as previous similar views but im not sure how they would happen for this view

Reply:
This was to basically  catch any unknown error and send reasonable response.

Reply:
I think they dont care about specific error. Message I will catch a generation Exception and send the error message as resonse.

Reply:
Yeah, what can even raise an `AttributeError` here? This catches a very specific class of error, not something generic, and one that may not even be possible in this code.

Reply:
Changed the code.